src/renderer/components/ScreenSharePicker.tsx

```diff
@@ -106,14 +115,24 @@ export function openScreenSharePicker(screens: Source[], skipPicker: boolean) {
                     modalProps={props}
                     submit={async v => {
                         didSubmit = true;
+
                         if (v.audioSource && v.audioSource !== "None") {
-                            if (v.audioSource === "Entire System") {
-                                await VesktopNative.virtmic.startSystem(v.workaround);
-                            } else {
-                                await VesktopNative.virtmic.start([v.audioSource], v.workaround);
+                            if (!v.audioDevice && v.audioSource && v.audioSource !== "None") {
+                                if (v.audioSource === "Entire System") {
+                                    await VesktopNative.virtmic.startSystem(v.workaround);
+                                } else {
+                                    await VesktopNative.virtmic.start([v.audioSource], v.workaround);
+                                }
                             }
+
+                            patchDisplayMedia({
+                                audioId: v.audioDevice,
+                                venmic: !!v.audioSource && v.audioSource !== "None",
+                                videoId: v.cameraId
+                            });
+
```

Comment:
Just switched over to Linux/Vesktop and was looking for a solution like this! 

I had to move patchDisplayMedia() outside of the above if-else block to get this to work. Still fails initially, but the streaming button afterwards will grab the then grab the camera every time.